src/test/ui/statics/static-promotion.rs

```diff
@@ -0,0 +1,23 @@
+// check-pass
+
+// Use of global static variables in literal values should be allowed for
+// promotion.
+// This test is to demonstrate the issue raised in
+// https://github.com/rust-lang/rust/issues/70584
+
+// Literal values were previously promoted into local static values when
+// other global static variables are used.
+
+struct A<T: 'static>(&'static T);
```

Comment:
This test does not actually test the thing you're trying to test. Your test passes on the playground

Reply:
Sorry, I pasted the wrong test case from the playground. :sweat:

Reply:
no worries